app/controllers/user_plans_controller.rb

```diff
@@ -0,0 +1,34 @@
+class UserPlansController < ApplicationController
+  before_action :check_amount, only: :create
+
+  def index
+  end
+
+  def create
+    customer = Stripe::Customer.create(email: params[:stripeEmail],
+                                       source: params[:stripeToken])
+
+    Stripe::Charge.create(
```

Comment:
You could extract this code block into a method and avoid using instance variable by doing something like this.

```ruby
def charge_customer_through_stripe(customer_id)
    amount, description = amount_and_description_by_plan
    Stripe::Charge.create(
       customer: customer_id,
       amount: amount,
       description: description,
      currency: "usd"
     )
end

def amount_and_description_by_plan
   if params[:plan] == "medial"
       [2000, "Workdey medial user"]
     elsif params[:plan] == "maestro"
        [3000, "Workdey maestro user"]
     end
  end
```